server/test/unit/BraveAlerterConfiguratorTest/getHistoricAlertsByAlertApiKeyTest.js

```diff
@@ -0,0 +1,148 @@
+// Third-party dependencies
+const { expect, use } = require('chai')
+const { afterEach, beforeEach, describe, it } = require('mocha')
+const sinon = require('sinon')
+const sinonChai = require('sinon-chai')
+
+const { helpers, HistoricAlert, ALERT_TYPE } = require('brave-alert-lib')
+const db = require('../../../db/db.js')
+const BraveAlerterConfigurator = require('../../../BraveAlerterConfigurator.js')
+
+// Configure Chai
+use(sinonChai)
+
+describe('BraveAlerterConfigurator.js unit tests: getHistoricAlertsByAlertApiKey and createHistoricAlertFromRow', () => {
+  beforeEach(() => {
+    this.maxTimeAgoInMillis = 60000
+    sinon.stub(helpers, 'getEnvVar').returns(this.maxTimeAgoInMillis)
+
+    const braveAlerterConfigurator = new BraveAlerterConfigurator()
+    this.braveAlerter = braveAlerterConfigurator.createBraveAlerter()
+  })
+
+  afterEach(() => {
+    helpers.getEnvVar.restore()
+    db.getHistoricAlertsByAlertApiKey.restore()
+  })
+
+  it('if there is a single result from db.getHistoricAlertsByAlertApiKey with num_presses > 1, returns an array containig a single HistoricAlert object with the returned data and ALERT_TYPE.BUTTONS_URGENT', async () => {
+    const results = {
+      id: 'id',
+      unit: 'unit',
+      incident_type: 'incidentType',
+      num_presses: 2,
+      created_at: new Date('2020-01-20T10:10:10.000Z'),
+      responded_at: new Date('2020-01-20T10:12:40.000Z'),
+    }
+    sinon.stub(db, 'getHistoricAlertsByAlertApiKey').returns([results])
+
+    const historicAlerts = await this.braveAlerter.getHistoricAlertsByAlertApiKey('alertApiKey', 'maxHistoricAlerts')
```

Comment:
I don't understand why the string `'maxHistoricAlerts'` is passed to `getHistoricAlertsByAlertApiKey` throughout this test file. shouldn't `maxHistoricAlerts` be an integer?

Reply:
ah ok, it's because of the stub.

Reply:
Just because it's not actually being used. This is a unit test for `getHistoricAlertsByAlertApiKey`, but all it does with its second parameter is pass it along to `db.getHistoricAlertsByAlertApiKey` which is stubbed out above to return the `results` array no matter what it's passed.

So I could pass it an integer, but it also wouldn't affect the test in any way. What do you think would be clearer from reading the test?

Reply:
I think if I had read the tests more closely I would have seen the stub and realized what was going on sooner. I did figure it out eventually.

you could make a `const maxHistoricAlerts = 'maxHistoricAlerts'` with a comment explaining what's going on, then reuse the constant throughout the tests - that would make it super clear - but I don't think it's necessary.

Reply:
OK, I'll leave it. Sorry about the confusion.